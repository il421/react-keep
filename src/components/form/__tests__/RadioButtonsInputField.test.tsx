import React from "react";
import { mount, ReactWrapper } from "enzyme";
import {
  RadioButtonsInputField,
  RadioButtonsInputFieldProps,
} from "../RadioButtonsInputField";
import { Form } from "react-final-form";

let props: RadioButtonsInputFieldProps,
  wrapper: ReactWrapper<RadioButtonsInputFieldProps, {}>;

beforeEach(() => {
  props = {
    name: "name",
    id: "id",
    radioClassName: "radioClassName",
  };
});

test("should render RadioButtonsInputField, and contain input element", () => {
  wrapper = mount<RadioButtonsInputFieldProps>(
    <Form initialValues={{}} onSubmit={() => {}}>
      {() => <RadioButtonsInputField {...props} />}
    </Form>
  );
  expect(wrapper.debug()).toMatchSnapshot();
  expect(wrapper.find("#id").exists()).toBeTruthy();
  expect(wrapper.find("label[htmlFor='id']").exists()).toBeTruthy();
});
