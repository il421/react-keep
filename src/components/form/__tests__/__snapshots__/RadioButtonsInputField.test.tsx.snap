// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RadioButtonsInputField, and contain input element 1`] = `
"<ReactFinalForm initialValues={{...}} onSubmit={[Function: onSubmit]}>
  <RadioButtonsInputField name=\\"name\\" id=\\"id\\" radioClassName=\\"radioClassName\\">
    <Field name=\\"name\\" id=\\"id\\" radioClassName=\\"radioClassName\\" type=\\"radio\\" component={[Function: FieldAdapter]}>
      <FieldAdapter input={{...}} meta={{...}} id=\\"id\\" radioClassName=\\"radioClassName\\" render={[undefined]}>
        <div className=\\"radioClassName\\">
          <input id=\\"id\\" name=\\"name\\" checked={true} onBlur={[Function]} onChange={[Function]} onFocus={[Function]} type=\\"radio\\" value={[undefined]} />
          <label htmlFor=\\"id\\" style={[undefined]} />
        </div>
      </FieldAdapter>
    </Field>
  </RadioButtonsInputField>
</ReactFinalForm>"
`;
